src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -56,12 +56,12 @@ private BoundExpression LowerDeclarationPattern(BoundDeclarationPattern pattern,
             if (pattern.IsVar)
             {
                 Debug.Assert(input.Type == pattern.LocalSymbol.Type);
-                var assignment = _factory.AssignmentExpression(_factory.Local(pattern.LocalSymbol), input);
+                var assignment = _factory.AssignmentExpression(pattern.VariableAccess, input);
                 var result = _factory.Literal(true);
                 return _factory.Sequence(assignment, result);
             }
 
-            return MakeDeclarationPattern(pattern.Syntax, input, pattern.LocalSymbol, requiresNullTest: true);
+            return MakeDeclarationPattern(pattern.Syntax, input, pattern.VariableAccess, requiresNullTest: true);
```

Comment:
Ditto.

---
In reply to: [77386808](https://github.com/dotnet/roslyn/pull/13535#discussion_r77386808) [](ancestors = 77386808)